### DISCUSSION TITLE
[Potential Bug?] highlighted bonds interact with eachother

### DISCUSSION THREAD

**Discussion by thwelln** [rdkit/rdkit#6967] [Q&A] [answered]
Hi there!

I'm currently using `rdMolDraw2D.PrepareAndDrawMolecule()` on the `MolDraw2DSVG`-drawer to highlight certain bonds (in different colors) and noticed that the shape of the highlighted bonds can change depending on other highlighted bonds, a behaviour that I cannot understand.

As an example I will use **barban**, (SMILES: **ClCC#CCOC(=O)Nc1cccc(c1)Cl**):
<img width="370" alt="image" src="https://github.com/rdkit/rdkit/assets/15370724/6a423746-9395-432e-ab03-57590076c110">

Highlighting bond 2 or 3 individually leads to the expected highlighting:
![image](https://github.com/rdkit/rdkit/assets/15370724/0ae5f8f3-69e0-4538-9eff-c78635b29240)

However, highlighting both together leads to this:
<img width="169" alt="image" src="https://github.com/rdkit/rdkit/assets/15370724/d18800af-a1dd-4ea6-b715-83cfd3ef746d">

Highlighting bond 1 and 2 together does not show that change:
<img width="169" alt="image" src="https://github.com/rdkit/rdkit/assets/15370724/911723c3-64f9-4cdb-aa37-597f56d4558b">

Can anyone explain what is going on?
Is this a property of the bonds that I am not aware of? Or is it similar to Natta Projection indicating the direction?

Bond 2 looks even weirder when all three are highlighted btw:
<img width="169" alt="image" src="https://github.com/rdkit/rdkit/assets/15370724/fa06a5a5-10be-4151-8361-0ba995ac8086">

The code I'm using boils down to this:
`d = rdMolDraw2D.MolDraw2DSVG(500, 500)`
`d.drawOptions().useBWAtomPalette()`
`rdMolDraw2D.PrepareAndDrawMolecule(d, m, highlightAtoms=[],`
`                                   highlightAtomColors={},`
`                                   highlightBonds=[2,3],`
`                                   highlightBondColors={2:(1,0,0),3:(1,0,0})`

I'm using RDKit version 2023.03.1.

Any help is welcome, I would really like to highlight the bonds "normally" for my masterthesis tool.
Feel free to ask for clarification if I ommited something important.

Thank you very much in advance!

**Comment by DavidACosgrove** [comment, marked as answer]
Hi @thwelln, 
Thanks for this. It is undoubtedly a bug.  What version of RDKit are you using?  
thanks,
Dave

**Reply by thwelln** [reply]
Hi Dave, thanks for answering this quick! I'm using version **2023.03.1** (as Python3 module).

**Comment by DavidACosgrove** [comment]
@thwelln, are you able to try it in 2023.09?  It looks similar to something I fixed in the last few months.

**Reply by thwelln** [reply]
I knew it'll one day come back to haunt me that I did not use a virtual environment. ;)
Unfortunately I cannot upgrade rdkit over pip right now, I get an error about a dll. But I'll try on another server and will keep you updated.

**Comment by DavidACosgrove** [comment]
Hang fire for a bit then. I’ll try it here later this evening. If it
doesn't work then that’ll save you some hassle.

**Reply by thwelln** [reply]
I just tested it on 2023.09.1. Unfortunately leads to the same problem. I also checked with MolDraw2DCairo in both 2023.03.1 and 2023.09.1 and the behaviour is visible in these as well.

**Comment by DavidACosgrove** [comment]
Ok, thanks for doing that. I’ll get it fixed in the next day or two.
